### PR TITLE
LPS-143988 Create PUT UserGroup by ExternalReferenceCode for User Group Headless API

### PR DESCRIPTION
**Pull request by erickmonteiroo**
@drewbrokke @pei-jung

### CODE REVIEW

**Comment by erickmonteiroo** [comment]
ci:stop

**Comment by erickmonteiroo** [comment]
ci:test:relevant

**Comment by erickmonteiroo** [comment]
ci:test:relevant

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/rest-openapi.yaml [hunk 1]
```diff
@@ -3878,6 +3878,35 @@ paths:
...
+                    content:
+                        application/json:
+                            schema:
+                                $ref: "#/components/schemas/UserGroup"
+                        application/xml:
+                            schema:
+                                $ref: "#/components/schemas/UserGroup"
+                    # @review
+                    description:
+                        "User group successfully replaced"
```
Comment (drewbrokke) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/rest-openapi.yaml:3908]: Brian will probably drop this description. PUT is expected to replace a whole entity so this doesn't add much.

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/jaxrs/exception/mapper/DuplicateUserGroupExceptionMapper.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+public class DuplicateUserGroupExceptionMapper
+	extends BaseExceptionMapper<DuplicateUserGroupException> {
+
+	@Override
+	protected Problem getProblem(
+		DuplicateUserGroupException duplicateUserGroupException) {
+
+		return new Problem(
+			Response.Status.BAD_REQUEST,
+			"A user group already exists with this name " +
```
Comment (drewbrokke) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/jaxrs/exception/mapper/DuplicateUserGroupExceptionMapper.java:46]: "A user group already exists with ~this~ *the* name "

**Review by drewbrokke — CHANGES_REQUESTED** [review]
@erickmonteiroo a couple of changes are needed. Also, please add test cases for the new remote and local service methods you added. Ping me if you want to talk over what to test.

Thanks!

**Comment by erickmonteiroo** [comment]
ci:test:sf

**Comment by erickmonteiroo** [comment]
ci:test:sf

**Comment by erickmonteiroo** [comment]
ci:test:relevant

**Comment by erickmonteiroo** [comment]
ci:test:relevant

**Comment by erickmonteiroo** [comment]
ci:test:relevant

**Comment by erickmonteiroo** [comment]
ci:test:relevant

**Comment by erickmonteiroo** [comment]
@epark18 Heyy, could you check if there are any related failures? Thanks!

**Comment by epark18** [comment]
@erickmonteiroo Hey, it doesn't look like any of the failures are related to your changes. Either the failures encountered here are failing across all PRs, or it is a flaky issue that seems to pop up every now and then.

File: modules/apps/user-groups-admin/user-groups-admin-test/src/testIntegration/java/com/liferay/user/groups/admin/service/test/UserGroupServiceTest.java [hunk 3]
```diff
@@ -64,6 +70,57 @@
...
+			RandomTestUtil.randomString(), null);
+
+		_addResourcePermission(
+			UserGroup.class.getName(),
+			String.valueOf(TestPropsValues.getCompanyId()), ActionKeys.UPDATE,
+			user);
+
+		_userGroupService.addOrUpdateUserGroup(
+			externalReferenceCode, RandomTestUtil.randomString(),
+			RandomTestUtil.randomString(), null);
```
Comment (drewbrokke) [inline modules/apps/user-groups-admin/user-groups-admin-test/src/testIntegration/java/com/liferay/user/groups/admin/service/test/UserGroupServiceTest.java:96]: @erickmonteiroo does this test case fail if you remove the `_addResourcePermission` statement on line 89?

**Comment by drewbrokke** [comment]
See https://github.com/liferay-usm/liferay-portal/pull/431

**Pull request closed without merging**